### DISCUSSION TITLE
k3s ctr images import without sudo

### DISCUSSION THREAD

**Discussion by up-to-you** [k3s-io/k3s#6103] [Q&A] [answered]
Security restrictions in a company doesn't allow to use registry (actually there are network restrictions), so we are trying to solve the issue via:

`docker save "$IMAGE" | ssh -C user@k3s-host 'ctr images import -'`

But `containerd` requires sudo, e.g. `sudo ctr images import -`.
The `containerd` itself has a tool https://github.com/containerd/nerdctl with rootless mode, but as far as i found `containerd` is integrated into `k3s`, such that there is no dedicated `systemctl` service.

We don't want to run whole k3s with `rootless` mode, but only to be able to import images.

This `sudoers` hack is restricted too:
`user ALL = (root) NOPASSWD: /usr/local/bin/ctr`

Are there possibilities to solve it ?
Thanks in advance!

**Comment by brandond** [comment, marked as answer]
Have you considered changing the permissions and ownership of the containerd socket so that your non-root user has access to i?
